openprescribing/frontend/price_per_unit/prescribing_breakdown.py

```diff
@@ -0,0 +1,97 @@
+import numpy
+
+from frontend.models import Presentation
+from matrixstore.db import get_db, get_row_grouper
+from matrixstore.matrix_ops import get_submatrix
+
+from .substitution_sets import get_substitution_sets
+
+
+def get_prescribing(generic_code, date):
+    """
+    For a given set of substitutable presentations (identified by
+    `generic_code`) get all prescribing of those presentations on the given
+    date
+
+    Return value is a dict of the form:
+        {
+            ...
+            bnf_code: (quantity_matrix, net_cost_matrix)
+            ...
+        }
+    """
+    bnf_codes = get_substitution_sets()[generic_code].presentations
+    db = get_db()
+    date_column = db.date_offsets[date]
+    date_slice = slice(date_column, date_column + 1)
+
+    results = db.query(
+        """
+        SELECT
+          bnf_code, quantity, net_cost
+        FROM
+          presentation
+        WHERE
+          bnf_code IN ({})
+        """.format(
+            ",".join("?" * len(bnf_codes))
+        ),
+        bnf_codes,
+    )
```

Comment:
Yes, I agree. And also with the comment below about almost identical queries appearing elsewhere. There's also very similar code powering the spending and NCSO APIs. I was holding off factoring these away though until I had a better sense of the different use cases we would have for it. I think once I've done Ghost Branded Generics (which is the last bit of code using `frontend_prescribing` I'll have a firm idea and it will be a relatively easy job to replace this code with a function in the matrixstore library.